### DISCUSSION TITLE
`highlightedSnippet()` does not work

### DISCUSSION THREAD

**Discussion by marcreichel** [spatie/laravel-site-search#16] [Q&A] [answered]
When rendering the search results and trying to print the `$hit->highlightedSnippet()` nothing gets returned but `$hit->snippet()` does return something (unhighlighted of course).

Am I missing something?

**Comment by carlosmfnt** [comment]
I'm also experiencing this, `highlightedSnippet()` does not return anything, but `snippet()` returns things that shouldn't be visible. When I'm searching something like "empleo", a simple string, it returns code from another part of the website which is completely unrelated, like in this case, where it returns something related to Axios:

![image](https://user-images.githubusercontent.com/56879478/140700847-ea36ab07-8ffb-42c9-90ce-b02a92e6953e.png)

I'm not sure why this is happening, but it seems that sometimes it captures this random code lines, and passes them as the snippet.

**Comment by marcreichel** [comment]
@carlosmfnt I had the same problem as you. I got rid of this by adding `script` to the ignored css selectors in the `site_search.ignore_content_by_css_selector` setting.
It does not explain why this content gets displayed though. Maybe the Meilisearch typo tolerant feature sees "empleo" somewhere in your CSRF token?! 😅

**Reply by freekmurze** [reply]
@carlosmfnt You could also use a custom indexer to customise the `getEntries` method, to filter out any unwanted things.

**Comment by marcreichel** [comment]
@freekmurze Any updates on this? Am I doing something wrong on how the `highlightSnippet()` should work? Do I need to configure something else?
Just converting my issue into a discussion is not really helpful at all though.

**Reply by freekmurze** [reply]
We set up the attributes that Meilisearch should highlight here: 
https://github.com/spatie/laravel-site-search/blob/1837d2c256e2f039d259902efef25704d1a5ecde/src/Drivers/MeiliSearchDriver.php#L85

So we should get highlighted results back from Meilisearch in a separate field. That's being done here: https://github.com/spatie/laravel-site-search/blob/c940600429a40d5ea798a9649db4710fce98f2a1/src/SearchResults/Hit.php#L43-L50

I currently have no time to investigate this further, but feel free to submit a PR should you find that the package is in error.

**Comment by marcreichel** [comment, marked as answer]
Fixed by #20! Thank you very much @SimonJulian!

**Comment by freekmurze** [comment]
@SimonJulian @marcreichel Are you using the package for a public facing search? Would love to see it in action in a real world site.

**Reply by marcreichel** [reply]
@freekmurze I do on my own website 👍🏽 https://marcreichel.dev/ (on a large desktop screen) you'll find the site search in the upper right corner (also accessible via CMD+K). I've paired the site search feature with a Livewire component and sprinkled some AlpineJS here and there. For small mobile screens I'm still working on a good looking and accessible solution.

**Reply by freekmurze** [reply]
Nice, looks good!

**Reply by marcreichel** [reply]
Thank you 🙂
